datadog_checks_dev/datadog_checks/dev/tooling/manifest_validator/common/validator.py

```diff
@@ -0,0 +1,273 @@
+    METRIC_TO_CHECK_EXCLUDE_LIST = {
+        'openstack.controller',  # "Artificial" metric, shouldn't be listed in metadata file.
+        'riakcs.bucket_list_pool.workers',  # RiakCS 2.1 metric, but metadata.csv lists RiakCS 2.0 metrics only.
+    }
+    METADATA_PATH = {"v1": "/assets/metrics_metadata", "v2": "/assets/integration/metrics/metadata_path"}
+    METRIC_PATH = {"v1": "/metric_to_check", "v2": "/assets/integration/metrics/check"}
+    PRICING_PATH = {"v1": "/pricing", "v2": "/pricing"}
+
+    def validate(self, check_name, decoded, _):
+        if not self.should_validate() or check_name == 'snmp' or check_name == 'moogsoft':
+            return
+
+        metadata_path = self.METADATA_PATH[self.version]
+        metadata_in_manifest = decoded.get_path(metadata_path)
+
+        # metric_to_check
+        metric_path = self.METRIC_PATH[self.version]
+        metric_to_check = decoded.get_path(metric_path)
+
+        pricing = decoded.get_path("/pricing") or []
+
+        if metric_to_check:
+            metrics_to_check = metric_to_check if isinstance(metric_to_check, list) else [metric_to_check]
+            if any(p.get('metric') in metrics_to_check for p in pricing):
+                return
+            for metric in metrics_to_check:
+                metric_integration_check_name = check_name
+                # snmp vendor specific integrations define metric_to_check
+                # with metrics from `snmp` integration
+                if check_name.startswith('snmp_') and not metadata_in_manifest:
+                    metric_integration_check_name = 'snmp'
+                if (
+                    not is_metric_in_metadata_file(metric, metric_integration_check_name)
+                    and metric not in self.METRIC_TO_CHECK_EXCLUDE_LIST
+                ):
+                    self.fail(f'  metric_to_check not in metadata.csv: {metric!r}')
+
+        elif metadata_in_manifest:
+            # if we have a metadata.csv file but no `metric_to_check` raise an error
+            metadata_file = get_metadata_file(check_name)
+            if os.path.isfile(metadata_file):
+                for _, row in read_metadata_rows(metadata_file):
+                    # there are cases of metadata.csv files with just a header but no metrics
+                    if row:
+                        self.fail('  metric_to_check not included in manifest.json')
+
+
+class IsPublicValidator(BaseManifestValidator):
+    PUBLIC_PATH = {"v1": "/is_public", "v2": "/display_on_public_website"}
+
+    def validate(self, check_name, decoded, fix):
+        correct_is_public = True
+        path = self.PUBLIC_PATH[self.version]
+        is_public = decoded.get_path(path)
+        if not isinstance(is_public, bool):
+            output = '  required boolean: is_public'
+
+            if fix:
+                decoded.set_path(path, correct_is_public)
+                self.fix(output, f'  new `is_public`: {correct_is_public}')
```

Comment:
Should the decoded.set_path call be done from within the `fix` method? Otherwise `fix` is more of a `echo_fixed` method?

Reply:
Yeah, seems that's the way fix was originally implemented, to record the changes for output at the end of the run.